### PR TITLE
Update README include required configuration

### PR DESCRIPTION
**Pull request by granteagon**
If you use `adapter: 'sails-postgresql'` instead of `module: postgresql` it breaks.  Thought it worth mentioning in the README.

### CODE REVIEW

**Comment by devinivy** [comment]
What you're referencing here is sails-specific configuration.  To configure the adapter, you don't have to tell the adapter its module name.  @particlebanana perhaps you can elucidate?

**Comment by granteagon** [comment]
Happy too.. will update shortly

**Comment by devinivy** [comment]
@granteagon it's a little confusing because this configuration in the readme is aimed for those who use waterline without sails.  The configuration formats work slightly differently.  Are there sails docs that are incorrect?

**Comment by granteagon** [comment]
I can just add another section that says "Example configuration for Sails JS users:" and add an example there if you like.

**Comment by devinivy** [comment]
I think that's a good idea.  Let's let @particlebanana or @tjwebb weigh-in.  Thanks for helping out!

**Comment by particlebanana** [comment]
Yeah I'm down for better documentation. Waterline connections and Sails connections are similar but not equal. Sails wraps them up a bit differently.

**Comment by particlebanana** [comment]
A good example of how connections are defined in Waterline can be seen [here](https://github.com/balderdashy/waterline/blob/master/example/express/express-example.js#L22-L50)

**Comment by granteagon** [comment]
That example does not work on my system.  Empty datasets are returned when I use the adapter.  I have to use module to make it work.  I updated my code to try the solution according the example you sent, and it doesn't talk to my database.

It's probably worth noting that I'm working with a legacy database that has all uppercase names for tables and columns, just not sure whether that's contributing to my particular issue or not.  However I have tried to connect to a normal all lowercase version of a table and it still doesn't work.

Bottom line, if I use `adapter:` sails-postgresql won't talk to my database, if I use `module:`, it works fine.

**Comment by granteagon** [comment]
This may be a bug, I'm trying to solve with documentation instead of code.  I'm just not familiar with this codebase enough to make that judgement.

**Comment by particlebanana** [comment]
Are you using Sails? In the latest version of Sails, `0.11.0`, the following config should work. Not sure why `module` would be a config key unless the ORM hook in Sails got changed. To handle custom table names and column names you can use the `tableName` key on a model to map it to a table other than the actual identity and the same for `columnName` on a model attribute.

```javascript
// config/connections.js
module.exports.connections = {
  someMongodbServer: {
    adapter: 'sails-mongo',
    host: 'localhost',
    port: 27017,
    user: 'username',
    password: 'password',
    database: 'your mongo db name here'
  },
  somePostgresqlServer: {
    adapter: 'sails-postgresql',
    host: 'YOUR_POSTGRES_SERVER_HOSTNAME_OR_IP_ADDRESS',
    user: 'YOUR_POSTGRES_USER',
    password: 'YOUR_POSTGRES_PASSWORD',
    database: 'YOUR_POSTGRES_DB'
  }
}
```

```javascript
// config/model.js
module.exports.models = {
  connection: 'somePostgresqlServer',
  migrate: 'alter',
  schema: true
}
```

If you are using Waterline as a standalone module there are a variety of ways you can configure it but using the [example](https://github.com/balderdashy/waterline/blob/master/example/express/express-example.js) in the repo is a good place to start.

**Comment by granteagon** [comment]
Yeah, I think this may be a genuine bug then.  Because as I mentioned in my original issue [#192], I have updated local sails, and global sails to their latest versions.

Here's my connections.js:
```js
var postgres = require('sails-postgresql'),
    mongo = require('sails-mongo'),
    db2 = require('sails-db2');

var connections = {

  default: 'hotData',

  adapters: {
    relational: postgres,
    documentStore: mongo,
    legacy: db2
  },

  hotData: {
    adapter: 'documentStore',
    host: 'localhost',
    port: 27017,
    user: 'hotuser',
    password: 'hotpass',
    database: 'hotdb'
  },

  // cold data / reports
  coldData: {
    module: 'sails-postgresql',
    host: 'localhost',
    port: 5432,
    user: 'colduser',
    password: 'coldpass',
    database: 'colddb'
  },

  legacyData: {
    adapter: 'legacy',
    host: 'localhost',
    port: 50000,
    user: 'db2user',
    password: 'db2pass',
    database: 'legacydb'
  }

};

module.exports.connections = connections;
```

If I change the postgres bit to this, it just returns empty result sets:
```js
  // cold data / reports
  coldData: {
    adapter: 'postgres',
    host: 'localhost',
    port: 5432,
    user: 'theuser',
    password: 'thepass',
    database: 'thecolddb'
  },
```

I'm using `tableName` to set the source table in the specific model js files to the correct name, but not the column names.  I will try setting a column name and see what I get back.  Will respond shortly.

**Comment by particlebanana** [comment]
Ah I think I know the problem. You are mixing up the two types of configs. What's happening is a fallback to sails-disk because it doesn't recognize a valid connection and sails-disk doesn't have any data in it. 

You need to make `config/connections.js` look like this:

```javascript
var connections = {

  hotData: {
    adapter: 'sails-mongo',
    host: 'localhost',
    port: 27017,
    user: 'hotuser',
    password: 'hotpass',
    database: 'hotdb'
  },

  // cold data / reports
  coldData: {
    adapter: 'sails-postgresql',
    host: 'localhost',
    port: 5432,
    user: 'colduser',
    password: 'coldpass',
    database: 'colddb'
  },

  legacyData: {
    adapter: 'sails-db2',
    host: 'localhost',
    port: 50000,
    user: 'db2user',
    password: 'db2pass',
    database: 'legacydb'
  }

};

module.exports.connections = connections;
```

And then set which connection is being used on each model if your models will use different databases. If they will all use the same database set it in `config/model.js` like so:

```javascript
// config/model.js
module.exports.models = {
  connection: 'coldData',
  migrate: 'alter',
  schema: true
}
```

Otherwise set the connection on each model like so:

```javascript
module.exports = {
  connection: 'coldData',
  attributes: {}
}
```

**Comment by granteagon** [comment]
Yeah, that's what I tried first, but that's not working either.  It only works when I use `module:` in place of `adapter:`.  It's weird.  I get that's it's probably falling back to sails-disk, which must be why it doesn't throw a config error, but returns an empty result set.  But for some reason I can only get it work with `module:`.

**Comment by particlebanana** [comment]
@granteagon if you still can't get it to work you can file a bug with Sails. The examples we posted above are tested and work though. If you replace your connections file with the code I sent it will work with the current versions of Sails and Waterline.
